tests/testthat/knitr_formats/test.qmd

```diff
@@ -0,0 +1,63 @@
+# Test #
+
+Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod
+tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At
+vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren,
+no sea takimata sanctus est Lorem ipsum dolor sit amet.
+
+```{r}
+a = 1
+```
+
+Test
+====
+
+Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod
+tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At
+vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren,
+no sea takimata sanctus est Lorem ipsum dolor sit amet.
+
+```{r}
+b <- function(x) {
+  d = 1
+}
+
+```
+
+```{r engine="python"}
+a=[]
+
+a[0]=1
+```
+
+```
+Plain code blocks can be written after three or more backticks
+- R Markdown: The Definitive Guide. Xie, Allaire and Grolemund (2.5.2)
+```
+
+```r
+# This is a non-evaluated block of R code for formatting in markdown.
+# It should not be linted
+abc = 123
+```
```

Comment:
Perhaps code blocks of the following forms would be worth including in the test?

````
```{.r}
1+1
```

```{{r}}
1+1
```
````

https://quarto.org/docs/output-formats/html-code.html#code-overflow
https://quarto.org/docs/computations/execution-options.html#unexecuted-blocks

Reply:
Great idea! Thanks.

I have added them in https://github.com/r-lib/lintr/pull/1487/commits/0b329127049a817bff7c4903e3c967b546622a6d.

These unexecuted blocks indeed don't produce a lint.